src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,125 @@
+package core.basesyntax;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidUserException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceTest {
+
+    private static final String VALID_LOGIN = "user12345";
+    private static final String VALID_PASSWORD = "qwerty123";
+    private static final int VALID_AGE = 20;
+
+    private static final String NULL_LOGIN = null;
+    private static final String SHORT_LOGIN = "user";
+    private static final String NULL_PASSWORD = null;
+    private static final String SHORT_PASSWORD = "pass";
+    private static final int NEGATIVE_AGE = -25;
+    private static final Integer NULL_AGE = null;
+    private static final int UNDERAGE = 17;
+
+    private RegistrationService registrationService;
+    private StorageDao storageDao;
+
+    @BeforeEach
+    public void setUp() {
+        storageDao = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl(storageDao);
+    }
+
+    @Test
+    public void register_validUser_registersSuccessfully() {
```

Comment:
please use the name pattern from the common mistakes:
```suggestion
    public void register_validUser_ok() {
```